osquery/tables/system/windows/bios_info.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+#include "osquery/core/windows/wmi.h"
+#include <osquery/tables.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+const auto kHPBiosSettingRegex = boost::regex("\\*([\\w ]*)");
+
+QueryData genBiosInfo(QueryContext& context) {
+  QueryData results;
+  std::string manufacturer;
+  WmiRequest wmiComputerSystemReq(
+      "select Manufacturer from Win32_ComputerSystem");
+  std::vector<WmiResultItem>& wmiComputerSystemResults =
+      wmiComputerSystemReq.results();
+  wmiComputerSystemResults[0].GetString("Manufacturer", manufacturer);
+  transform(manufacturer.begin(),
+            manufacturer.end(),
+            manufacturer.begin(),
+            ::tolower);
+  if ((manufacturer == "hp") || (manufacturer == "hewlett-packard") ||
```

Comment:
I agree that this probably isn't the best way to go about it. Just so I'm clear, should I may maps for each name that the manufacturer uses? For instance, some machines have HP, while other will use Hewlett-Packard, so just matching on one won't provide all results. My maps would look something like this:

{"HP":{"select Name,Value from HP_BiosSetting","root\\hp\\instrumentedBIOS"}}
{"Hewlett-Packard":{"select Name,Value from HP_BiosSetting","root\\hp\\instrumentedBIOS"}}
{"Hewlett Packard":{"select Name,Value from HP_BiosSetting","root\\hp\\instrumentedBIOS"}}

I still need to then check the vendor since the WMI classes that vendors provide are different. 

Another option would be to create a helper method to sanitize the vendors before I do any of this, that way we only need to store one vendor map. I could pass in something like Hewlett-Packard or HP into the method and always get back hp as the vendor. Thinking about it, I could also just use this method to return the preferred map.

Thoughts?